src/engine/source/api/src/catalog/catalog.cpp

```diff
@@ -164,6 +167,17 @@ std::optional<base::Error> Catalog::postResource(const Resource& collection, con
                                         Resource::typeToStr(Resource::Type::collection))};
     }
 
+    Format contentFormat;
+    try
+    {
+        auto tmp = json::Json {content.c_str()};
+        contentFormat = Format::json;
+    }
+    catch (const std::exception& e)
+    {
+        contentFormat = Format::yaml;
+    }
```

Comment:
How can you be sure that if it is not json it is a valid yaml? You shouldn't auto deduce the type, it's information you already have.